pkg/objectstore/objectstore.go

```diff
@@ -0,0 +1,100 @@
+package objectstore
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/glebarez/sqlite"
+	"gorm.io/gorm"
+	"k8s.io/utils/strings/slices"
+)
+
+var ErrUnregisteredType = func(t interface{}) error { return fmt.Errorf("%T is not registered with objectstore", t) }
+
+type DB struct {
+	location        string
+	db              *gorm.DB
+	registeredTypes []string
+}
+
+type DBConfig struct {
+	Location string
+}
+
+func NewObjectStore(options ...Option) *DB {
+	config := DBConfig{
+		Location: ":memory:",
+	}
+
+	for _, opt := range options {
+		opt(&config)
+	}
+
+	return &DB{
+		location:        config.Location,
+		registeredTypes: make([]string, 8), //nolint:gomnd
```

Comment:
Should this be `make([]string, 0, 8)`? Else the append statement in `Connect` is going to append new values to a slice of 8 empty strings.